tools/mason/MasonPublish.chpl

```diff
@@ -0,0 +1,248 @@
+        }
+      }
+      else {
+        throw new owned MasonError('Must have remote origin set up in repository in order to publish.');
+      }
+    }
+    else throw new owned MasonError('Usage must meet the form "mason publish [options] <username>"');
+  }
+  catch e: MasonError {
+    writeln(e.message());
+    exit(1);
+  }
+}
+
+/* Main Script that goes through the act of publishing the package to the mason registry.
+ Takes the package owners GitHub usrename as input will throw errors through command
+line git commands if any of the git calls fails.*/
+proc publishPackage(username: string) throws {
+  try! {
+    const packageLocation = getEnv('PWD');
+    here.chdir(MASON_HOME);
+    const dir =  getEnv('PWD');
+    mkdir('tmp');
+    here.chdir(MASON_HOME + '/tmp');
+    cloneMasonReg(username);
+    here.chdir(packageLocation);
+    const name = getName();
+    branchMasonReg(username, name);
+    const cwd = getEnv("PWD");
+    const package=  addPackageToBricks(packageLocation);
+    here.chdir(MASON_HOME + "/tmp/mason-registry");
+    const projectHome = getProjectHome(cwd);
+    runCommand("git add .");
+    runCommand("git commit -m '" + package +"'");
+    runCommand('git push --set-upstream origin ' + package, true);
+    here.chdir(MASON_HOME);
+    rmTree('tmp');
+    writeln('--------------------------------------------------------------------');
+    writeln('Go to the above link to open up a Pull Request to the mason-registry');
+  }
+  catch {
+    here.chdir(MASON_HOME);
+    rmTree('/tmp/');
+    writeln('Error publishing your package to the mason-registry');
+  }
+}
+
+/* If --dry-run is passed then it takes the username and checks to see if the mason-registry is forked
+ and the package has a git remote origin. If both exist then the package can be published. */
+proc dryRun(username: string) throws {
+  var fork = false;
+  var checkIfForkExists = ('git ls-remote https://github.com/' + username + '/mason-registry').split();
+  var p = spawn(checkIfForkExists, stdout=PIPE);
+  p.wait();
+  if p.exit_status == 0 {
+    fork = true;
+  }
+  var git = false;
+  if doesGitOriginExist() {
+    git = true;
+  }
+  if git && fork {
+    writeln('Package can be published to the mason-registry');
+    writeln('Commands that will ran are:');
+    writeln('git clone git:github.com:[username]/mason-registry mason-registry');
+    writeln('git checkout -b [package name]');
+    writeln('Package Name will be added to the Bricks in the mason-registry');
+    writeln('git add .');
+    writeln('git commit -m [package name]');
+    writeln('git push --set-upstream origin [package name]');
+  }
+  else {
+    if fork == false {
+      throw new owned MasonError('mason-registry is not forked on your GitHub');
+    }
+    else {
+      throw new owned MasonError('Package does not gave a git origin');
+    }
+  }
+}
+
+
+proc moveToMasonHome() {
+  const masonHome = getEnv("MASON_HOME");
```

Comment:
`MasonEnv.chpl` already defined a `MASON_HOME` - let's use that.

Reply:
and in that case, this function becomes `here.chdir(MASON_HOME)`, so we might as well remove this function.